### DISCUSSION TITLE
Review required, but required approvals == 0

### DISCUSSION THREAD

**Discussion by FIASJason** [community/community#74977] [New to GitHub] [answered]
I've created an organisation-wide branch protection rule to require a pull request before merging. At some point, I probably set the number of required approvals to 1, but it's since been reset to 0.

One of our repositories is still forcing us to complete a code review despite the `Required approvals` property being set to 0.

![image](https://github.com/community/community/assets/145747752/abe394c2-c1a2-426e-a5ab-1684f579bc1c)

![image](https://github.com/community/community/assets/145747752/451fb420-323f-4f38-a811-3edda449bcda)

It's possible that I've missed something, but my understanding is that this rule will prevent contributors from pushing directly to `main` (the protected branch) without a code review.

How can I modify the rule to allow people with `Write` permissions to merge pull requests without requiring a code review?

**Comment by MegatIrfan** [comment, marked as answer]
Try This : 

**1) Navigate to Repository Settings:**
Go to your repository on the platform (e.g., GitHub).
Navigate to the "Settings" or "Repository settings" section.

**2) Branches Section:**
Look for a section related to "Branches" or "Branch protection rules."

**3) Modify Protection Rule:**
Find the branch for which you want to modify the protection rule (e.g., "main").
If there's an existing protection rule, click on it to edit the settings.

**4) Adjust Required Approvals:**
Look for the setting related to "Required approvals" or similar.
Set the number of required approvals to the desired value (e.g., 0).

**5)Other Settings:**
Check for other settings related to code reviews or branch protection that might affect the behavior.



Save your changes.

**Reply by FIASJason** [reply]
🤦‍♂️ Thank you so much, @MegatIrfan ! I was certain I'd looked everywhere, but your reply helped me find the culprit.

I hope someone else finds this useful, so I'll document the resolution here.

As in my original post, I'd checked the rulesets both for the repository and the organisation as a whole (which enforces rules on each repo in the organisation). I'd verified that the default ruleset required a pull request but no code reviews.

The `Rulesets` section showed that the number of required approvals was `0`:
![image](https://github.com/community/community/assets/145747752/6d259490-a13d-43d9-a652-7aaafad75a73)

#### However

There's also a `branches` section, just above the rulesets, which includes a single branch protection rule. 
![image](https://github.com/community/community/assets/145747752/da36b459-4c40-4625-9690-dd44a4c7ba7e)

**This rule requires 1 approval before merging** and appears to be set at the repo level.
![image](https://github.com/community/community/assets/145747752/7799fb67-4bf5-4550-8c33-9d596271717c)

I've removed that rule, and voila! No more review approvals.

**Reply by MegatIrfan** [reply]
I'm glad to hear that you found the resolution to the issue :)

**Reply by queenofcorgis** [reply]
@FIASJason @MegatIrfan appreciate you both documenting the steps and sharing them so it can help others - awesome work!

**Comment by beattobee** [comment]
DApps are often built using smart contract to execute the functions of the application and to maintain records on the blockchain.

**Reply by npty** [reply]
how does this related? 😅
